extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxRecorder.java

```diff
@@ -80,10 +82,16 @@ void registerMessageConsumers(Map<String, ConsumeEvent> messageConsumerConfigura
                 } else {
                     consumer = eventBus.consumer(address);
                 }
+
+                // Create a context attached to each consumer
+                // If we don't all consumers will use the same event loop and so published messages (dispatched to all
+                // consumers) delivery is serialized.
+                EventLoopContext context = ((VertxInternal) vertx).createEventLoopContext();
                 consumer.handler(new Handler<Message<Object>>() {
                     @Override
                     public void handle(Message<Object> m) {
                         if (invoker.isBlocking()) {
+                            // If we are blocking, do not use the captured context.
                             vertx.executeBlocking(new Handler<Promise<Object>>() {
```

Comment:
I think here instead of doing executeBlocking you could rely on the `Context` created before and create instead a worker with `VertxInternal#createWorkerContext()`